src/main/webapp/2Dscript.js

```diff
@@ -80,7 +175,418 @@ function addCenterControl(controlDiv) {
+  // clicked.
+  google.maps.event.addListener(subPath, 'click', linkTo3D);
+  subPath.setMap(map);
+  markerBottom = genMarker(subSectionData.minLatPoint.lat,
+      subSectionData.minLatPoint.lng);
+  markerTop = genMarker(subSectionData.maxLngPoint.lat,
+      subSectionData.maxLngPoint.lng);
+  markerBottom.setMap(map);
+  markerTop.setMap(map);
+
+  sessionStorage.setItem(id + '_' + type + '_' + subSectionNumber,
+      JSON.stringify(subLine));
+}
+
+/**
+ * Change the rectangle width to be match the current cursor position.
+ * @param {MouseEvent} event
+ */
+function genChangingBox(event) {
+  if (isMouseDown) {
+    console.log('Set width triggered');
+    finX = event.pageX;
+    finY = event.pageY;
+    widget.width(finX - x);
+    widget.height(finY - y);
+    widget.css({
+      'width': (finX - x) + '!important',
+      'height': (finY - y) + '!important',
+      'display': 'block',
+      'border': '2px dashed #ccc',
+    });
+  }
+}
+
+/**
+ * Generate a Google maps marker at the given lat and long.
+ * @param {number} latitude
+ * @param {number} longitude
+ * @return {google.maps.Marker}
+ */
+function genMarker(latitude, longitude) {
+  const marker = new google.maps.Marker({
+    position: {lat: latitude, lng: longitude},
+    title: 'Lat: ' + latitude + ' Lng: ' + longitude,
+  });
+  return marker;
+}
+
+
+/**
+ * Create a polyline from the given points with specified parameters.
+ * @param {Point} linePoints
+ * @param {string} color
+ * @param {number} opacity
+ * @param {number} weight
+ * @param {number} index
+ * @return {google.maps.Polyline}
+ */
+function getPolyLine(linePoints, color, opacity, weight, index = 1) {
+  const polyLine = new google.maps.Polyline({
+    path: linePoints,
+    geodesic: true,
+    strokeColor: color,
+    strokeOpacity: opacity,
+    strokeWeight: weight,
+    zIndex: index,
+  });
+  return polyLine;
+}
+/**
+ * @typedef {Object<Array<Point>, Point, Point>} subSectionData
+ * @property {Array<Point>} subLine
+ * @property {Point} minLatPoint
+ * @property {Point} maxLngPoint
+ */
+/**
+ * Compute the subsection of a line contained in a bounding rectangle.
+ * @param {Array<Point>} pose
+ * @param {number} currentLat
+ * @param {number} priorLat
+ * @param {number} currentLng
+ * @param {number} priorLng
+ * @return {subSectionData}
+ */
+function computeSubSection(pose, currentLat, priorLat, currentLng, priorLng) {
+  const minLat = Math.min(currentLat, priorLat);
+  const maxLat = Math.max(currentLat, priorLat);
+  const minLng = Math.min(currentLng, priorLng);
+  const maxLng = Math.max(currentLng, priorLng);
+  console.log(pose);
+  // Lat can be from [-90,90].
+  let discoveredMinLat = 91;
+  let discoveredMinLatPair = 181;
+  // Lng can be from [-180,180].
+  let discoveredMaxLng = -181;
+  let discoveredMaxLngPair = -91;
+  subLine = [];
+  for (point of pose) {
+    const loopLat = point.lat;
+    const loopLng = point.lng;
```

Comment:
I don't think these variable carry their weight. Just refer to `point.lat` and `point.lng` within the loop, so the reader doesn't have to keep another two variables in mind.